### PR TITLE
re-add `rebar3 nix init`

### PR DESCRIPTION
**Pull request by dlesl**

### CODE REVIEW

File: tests/shell.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{ pkgs ? import <nixpkgs> { } }:
```
Comment (gleber) [inline tests/shell.nix:1]: This does not work for me on my NixOS machine as I seem to use a version of nixpkgs without the `rebar3WithPlugins` yet. Can you pin it to a specific version? I typically use Niv for this, but a plain nix-based pin would work as well.

**Comment by gleber** [comment]
Thanks for setting up the pinning. I've tried the tests and it fails:

```
[gleber@luna:~/code/rebar3_nix/tests]$ git log -1
commit 5c1bf80bdd377ba4882f1db56e349f9f948851b5 (HEAD -> init-new, dlesl/init-new)
Author: David <dlesl@users.noreply.github.com>
Date:   Fri May 28 18:45:38 2021 +0200

    command line to mirror `rebar3 new`

[gleber@luna:~/code/rebar3_nix/tests]$ nix-shell -p gnumake --run make
...
[snip]
...
error: --- Error --- nix-daemon
builder for '/nix/store/l81xpf4n3wvbcbji7zlgj4r894a2jj4w-cowboy-2.8.0.drv' failed with exit code 1; last 10 log lines:
  Making symlink from /nix/store/inac1w93qn95fprizc98rq3g18bc756m-erlang-22.3/lib/erlang/lib/erl_interface-3.13.2 to _build/default/lib/erl_interface
  Making symlink from /nix/store/inac1w93qn95fprizc98rq3g18bc756m-erlang-22.3/lib/erlang/lib/edoc-0.11 to _build/default/lib/edoc
  Making symlink from /nix/store/inac1w93qn95fprizc98rq3g18bc756m-erlang-22.3/lib/erlang/lib/diameter-2.2.3 to _build/default/lib/diameter
  Making symlink from /nix/store/inac1w93qn95fprizc98rq3g18bc756m-erlang-22.3/lib/erlang/lib/xmerl-1.3.23 to _build/default/lib/xmerl
  Making symlink from /nix/store/inac1w93qn95fprizc98rq3g18bc756m-erlang-22.3/lib/erlang/lib/ssh-4.9 to _build/default/lib/ssh
  Making symlink from /nix/store/dacczdqhrj2dhyajnqj1yhsqr62rbww7-ranch-1.7.1/lib/erlang/lib/ranch-1.7.1 to _build/default/lib/ranch
  Making symlink from /nix/store/623jahc0hsdzadjs54nd2sq70wvfyx14-cowlib-2.9.1/lib/erlang/lib/cowlib-2.9.1 to _build/default/lib/cowlib
  building
  ===> Verifying dependencies...

error: --- Error ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-build
1 dependencies of derivation '/nix/store/drbfdfkvd9gavxyv4d9w6v7ai60h6dmj-test_app-0.1.0.drv' failed to build
make[1]: *** [Makefile:18: test_app] Error 1
make[1]: Leaving directory '/home/gleber/code/rebar3_nix/tests'
make: *** [Makefile:2: all] Error 2
```

**Comment by dlesl** [comment]
Ah, I didn't account for the nested nix-shell/nix-build calls. Could you try now?

**Comment by gleber** [comment]
Awesome changes! Thanks for the actions config

**Pull request merged**